### PR TITLE
Adding documention for hashing and masking using the transform processor

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
## Purpose of this pull request

This pull request (PR) adds documention for hashing and masking using the transform processor in opentelemetry

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions and updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, and new packages for the site, Gatsby, React, etc

### CODE REVIEW

**Comment by jpipkin1** [comment]
@rnishtala-sumo, I see that this PR is currently in draft. When you are ready for us to review it, click **Ready for review**.

**Review by jpipkin1 — APPROVED** [review]
Approved, @rnishtala-sumo. I made some small formatting updates to your new article, and I added the article to the left navigation bar (see commit [Misc updates](https://github.com/SumoLogic/sumologic-documentation/pull/3217/commits/6be7d688d03534fa78baefccc9ed08742334e6ac)).

File: docs/send-data/opentelemetry-collector/data-transformations.md
Comment (c-kruse) [inline docs/send-data/opentelemetry-collector/data-transformations.md:10]: Do you intend to link to your personal fork?

File: docs/send-data/opentelemetry-collector/data-transformations.md
Comment (c-kruse) [inline docs/send-data/opentelemetry-collector/data-transformations.md:16]: The inconsistent password regex between line 21 and 30 (`\w+`, `test`) had me a little confused.

File: docs/send-data/opentelemetry-collector/data-transformations.md
Comment (c-kruse) [inline docs/send-data/opentelemetry-collector/data-transformations.md:45]: Am I missing some material difference between example 1 and 2? Seems like a lot of duplicated content just to communicate that this hashing pattern can be applied to more than passwords.

File: docs/send-data/opentelemetry-collector/data-transformations.md
Comment (rnishtala-sumo) [reply docs/send-data/opentelemetry-collector/data-transformations.md:10]: ah yes, need to change this, thanks

File: docs/send-data/opentelemetry-collector/data-transformations.md
Comment (rnishtala-sumo) [reply docs/send-data/opentelemetry-collector/data-transformations.md:16]: yeah one extracts the match group into an attribute called "password", the second one looks for the password in the message.

Maybe a better idea to change the second regex to something more generic like "password=([0-9A-Za-z]+_)"

File: docs/send-data/opentelemetry-collector/data-transformations.md
Comment (rnishtala-sumo) [reply docs/send-data/opentelemetry-collector/data-transformations.md:45]: I wanted to illustrate two different examples. I could change the hash function in the second one to add more value.

Comment (c-kruse) [reply docs/send-data/opentelemetry-collector/data-transformations.md:45]: Makes sense!

Maybe hashing an attribute - just so that nobody gets confused about what hashing's got to do with the whole extract/hash/concat/replace contraption?

Comment (rnishtala-sumo) [reply docs/send-data/opentelemetry-collector/data-transformations.md]: made this change.

**Review by c-kruse — APPROVED** [review]
👍 👍

**Comment by jpipkin1** [comment]
@rnishtala-sumo, is this PR ready to publish yet? If so, you can click the "Merge when ready" button.

**Comment by rnishtala-sumo** [comment]
@jpipkin1 my local build is failing ATM with the following error
```Redirect file creation error for "/Users/rnishtala/src/sumologic-documentation/build/Search/index.html"```

taking one last look before I hit merge

**Comment by jpipkin1** [comment]
@rnishtala-sumo, please hold off on merging for now. We should investigate your error.

@kimsauce, this might be one of those PRs that could break our Search.

**Comment by kimsauce** [comment]
> @rnishtala-sumo, please hold off on merging for now. We should investigate your error.
> 
> @kimsauce, this might be one of those PRs that could break our Search.
> 
> When I Slacked with @rnishtala-sumo, he said:
> 
> > doing a `yarn build` fails with this error
> 
> ```
> [ERROR] Redirect file creation error for "/Users/rnishtala/src/sumologic-documentation/build/Search/index.html".
> [ERROR] Unable to build website for locale en.
> [ERROR] Error: The redirect plugin is not supposed to override existing files.
> ```

This can usually be fixed with a `yarn clear`, which clears your local cache. I just pulled this PR into my local and did that and `yarn install` just in case! Will try merging now and will monitor.

**Pull request merged**
